packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -1812,6 +1915,9 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
       bottom: _resizeToAvoidBottomInset ? mediaQuery.viewInsets.bottom : 0.0,
     );
 
+    // extendBody locked when keyboard is open
+    final bool _extendBody = minInsets.bottom > 0 ? false : widget.extendBody;
```

Comment:
i wonder if this is really correct in all cases...

Reply:
The value of `minInsets.bottom` is just the media query's value of  viewInsets.bottom (unless resizeToAvoidBottomInsets is false). It doesn't guarantee that the keyboard is up however if someone were to configure a MediaQuery that way we'd react the same was as if the keyboard is up.

I think this is OK, albeit complicated: we don't extend the bottom of the body if resizeToAvoidBottomInsets is true (the default) and viewInsets.bottom is non-zero.